Cargo.toml

```diff
@@ -18,3 +18,7 @@ html5ever = "0.26.0"
 markup5ever_rcdom = "0.2.0"
 serde = { version = "1.0", optional = true, features = ["derive"] }
 serde_json = "1.0"
+
+[dev-dependencies]
```

Comment:
These are very heavy dependencies for some trivial testing. I think we can get away with a raw TcpSocket.
Something like:
```
#[cfg(test)]
mod tests {
    use super::*;
    use std::net::TcpListener;
    use std::io::{Read, Write};

    #[test]
    fn test_headers() {
        let socket = TcpListener::bind("127.0.0.1:0").unwrap();  // bind to a random port
        let url = format!("{}", socket.local_addr().unwrap());
        std::thread::spawn(move || {
            let my_headers: Vec<String> = vec!["X-My-Header: 1234".to_string()];
            let options = WebpageOptions { headers: my_headers, ..Default::default()};
            let webpage = Webpage::from_url(&url, options);
            assert!(webpage.is_ok());
        });
        let mut stream = socket.accept().unwrap().0;
        let mut buf = vec![0; 1024];
        let mut read = 0;
        let mut request;
        loop {
            let bytes = stream.read(&mut buf[read..]).unwrap();
            assert_ne!(bytes, 0);
            read += bytes;
            request = String::from_utf8(buf[..read].to_vec()).unwrap();
            if request.contains("\r\n\r\n") {
                break
            }
        }
        assert!(request.contains("X-My-Header: 1234\r\n"));
        stream.write_all(b"HTTP/1.1 200 OK\r\n\r\n").unwrap();
    }
}
```

It's extremely rough around the edges but it saves a lot of dev-build time :)